src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -523,8 +538,16 @@ private SingleProjectInfo GetProjectInfo(LockFileLibrary library)
                 throw new BuildErrorException(Strings.CannotFindProjectInfo, library.Name);
             }
 
-            string mainProjectDirectory = Path.GetDirectoryName(_mainProjectInfo.ProjectPath);
-            string fullProjectPath = Path.GetFullPath(Path.Combine(mainProjectDirectory, projectPath));
+            string fullProjectPath;
+            if (_mainProjectInfo != null)
+            {
+                string mainProjectDirectory = Path.GetDirectoryName(_mainProjectInfo.ProjectPath);
+                fullProjectPath = Path.GetFullPath(Path.Combine(mainProjectDirectory, projectPath));
+            }
+            else
+            {
+                fullProjectPath = Path.GetFullPath(projectPath);
```

Comment:
Again, a test would be nice here - have a project that sets `<IncludeMainProjectInDepsFile>false</IncludeMainProjectInDepsFile>` and has a P2P reference.

I think this kind of scenario could be used in situations like the dotnet/cli uses.  See [https://github.com/dotnet/cli/blob/master/src/redist/redist.csproj#L62-L64](https://github.com/dotnet/cli/blob/master/src/redist/redist.csproj#L62-L64) where we have to go and strip the "main project" from the deps file.

---
In reply to: [111982912](https://github.com/dotnet/sdk/pull/1128#discussion_r111982912) [](ancestors = 111982912)